core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,876 @@
+            .withDefaultNamespace(viewIdentifier.namespace())
+            .withQuery("spark", "select * from ns.tbl")
+            .create();
+
+    assertThat(view).isNotNull();
+    assertThat(catalog().viewExists(tableIdentifier)).isFalse();
+    assertThat(catalog().viewExists(viewIdentifier)).isTrue();
+
+    Assertions.assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(viewIdentifier)
+                    .withSchema(OTHER_SCHEMA)
+                    .withQuery("spark", "select * from ns.tbl")
+                    .create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("View already exists: ns.view");
+
+    catalog().buildTable(tableIdentifier, SCHEMA).create();
+    assertThat(catalog().tableExists(tableIdentifier)).isTrue();
+    assertThat(catalog().viewExists(tableIdentifier)).isFalse();
+
+    Assertions.assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(tableIdentifier)
+                    .withSchema(OTHER_SCHEMA)
+                    .withDefaultNamespace(tableIdentifier.namespace())
+                    .withQuery("spark", "select * from ns.tbl")
+                    .create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("Table with same name already exists: ns.table");
+  }
+
+  @Test
+  public void renameView() {
+    TableIdentifier from = TableIdentifier.of("ns", "view");
+    TableIdentifier to = TableIdentifier.of("ns", "renamedView");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(from.namespace());
+    }
+
+    catalog()
+        .buildView(from)
+        .withSchema(SCHEMA)
+        .withDefaultNamespace(from.namespace())
+        .withQuery("spark", "select * from ns.tbl")
+        .create();
+
+    assertThat(catalog().viewExists(from)).as("View should exist").isTrue();
+    assertThat(catalog().listViews(from.namespace())).containsExactly(from);
+
+    catalog().renameView(from, to);
+
+    assertThat(catalog().listViews(to.namespace())).containsExactly(to);
+    assertThat(catalog().viewExists(from)).as("View should not exist").isFalse();
+    assertThat(catalog().viewExists(to)).as("View should exist").isTrue();
+
+    View view = catalog().loadView(to);
+    assertThat(view).isNotNull();
+
+    // validate view settings
+    assertThat(view.name()).isEqualTo(catalog().name() + "." + to);
+    assertThat(view.properties()).isEmpty();
+    assertThat(view.history())
+        .hasSize(1)
+        .first()
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(1);
+    assertThat(view.schema().asStruct()).isEqualTo(SCHEMA.asStruct());
+    assertThat(view.schemas()).hasSize(1).containsKey(SCHEMA.schemaId());
+    assertThat(view.versions()).hasSize(1).containsExactly(view.currentVersion());
+
+    ViewVersion viewVersion = view.currentVersion();
+    assertThat(viewVersion).isNotNull();
```

Comment:
Is it possible to create an expected version and have just one assertion?

Reply:
good idea, i've done that here and in a few other places

Reply:
While doing this refactoring for `renameViewUsingDifferentNamespace()` I've noticed that the `defaultNamespace` of the current view version was still set to the old namespace, which is wrong. 
I think we should disallow renaming a view if the namespaces are different of the `from` / `to` identifiers. Thoughts on this?

Reply:
> I think we should disallow renaming a view if the namespaces are different of the from / to identifiers. Thoughts on this?

Should be same as rename table behavior I guess. (Which will allow renaming tables to different namespaces)

Reply:
nvm, I've reverted those changes and renaming view will behave the same as renaming tables